packages/storage-plus/src/map.rs

```diff
@@ -260,6 +260,41 @@ where
     }
 }
 
+#[cfg(feature = "iterator")]
+impl<'a, K, T> Map<'a, K, T>
+where
+    T: Serialize + DeserializeOwned,
+    K: PrimaryKey<'a>,
+{
+    /// Clears the map, removing all elements.
+    pub fn clear(&self, store: &mut dyn Storage) {
+        const TAKE: usize = 10;
+        let prefix = self.no_prefix_raw();
+        let mut cleared = false;
+
+        while !cleared {
+            let paths = prefix
+                .keys_raw(store, None, None, cosmwasm_std::Order::Ascending)
```

Comment:
Just nitpicking, but IMO it's better to keep it in one place in this case.

```suggestion
        let mut cleared = false;

        while !cleared {
            let paths = self
                .no_prefix_raw()
                .keys_raw(store, None, None, cosmwasm_std::Order::Ascending)
```